File: flutter_project/lib/features/organizations/data/models/organization_member_model.dart
/// Negocio Flex - Organization Member Model (Flutter / Dart)

import '../../domain/entities/organization_entity.dart';
import '../../domain/entities/organization_member_entity.dart';

class OrganizationMemberModel extends OrganizationMemberEntity {
  const OrganizationMemberModel({
    required super.id,
    required super.organizationId,
    required super.userId,
    required super.role,
    super.status,
    super.userFullName,
    super.userEmail,
    super.userAvatarUrl,
    required super.createdAt,
    required super.updatedAt,
  });

  factory OrganizationMemberModel.fromJson(Map<String, dynamic> json) {
    OrganizationRole parseRole(String? role) {
      switch (role) {
        case 'owner': return OrganizationRole.owner;
        case 'admin': return OrganizationRole.admin;
        default: return OrganizationRole.staff;
      }
    }

    MemberStatus parseStatus(String? status) {
      switch (status) {
        case 'inactive': return MemberStatus.inactive;
        case 'invited': return MemberStatus.invited;
        case 'suspended': return MemberStatus.suspended;
        default: return MemberStatus.active;
      }
    }

    final rawProfiles = json['profiles'] as Map<String, dynamic>? ?? {};

    return OrganizationMemberModel(
      id: json['id'] ?? '',
      organizationId: json['organization_id'] ?? json['organizationId'] ?? '',
      userId: json['user_id'] ?? json['userId'] ?? '',
      role: parseRole(json['role']),
      status: parseStatus(json['status']),
      userFullName: rawProfiles['full_name'] ?? json['user_full_name'],
      userEmail: rawProfiles['email'] ?? json['user_email'],
      userAvatarUrl: rawProfiles['avatar_url'] ?? json['user_avatar_url'],
      createdAt: json['created_at'] != null ? DateTime.parse(json['created_at']) : DateTime.now(),
      updatedAt: json['updated_at'] != null ? DateTime.parse(json['updated_at']) : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'organization_id': organizationId,
      'user_id': userId,
      'role': role.name,
      'status': status.name,
      'user_full_name': userFullName,
      'user_email': userEmail,
      'user_avatar_url': userAvatarUrl,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }
}

File: src/features/organizations/data/models/organization_member_model.ts
/**
 * Negocio Flex - Organization Member Model (Fase 4)
 */

import {
  OrganizationMemberEntity,
  OrganizationRole,
  MemberStatus,
} from '../../domain/entities/organization_entity';

export class OrganizationMemberModel implements OrganizationMemberEntity {
  constructor(
    public readonly id: string,
    public readonly organizationId: string,
    public readonly userId: string,
    public readonly role: OrganizationRole,
    public readonly status: MemberStatus,
    public readonly createdAt: string,
    public readonly updatedAt: string,
    public readonly userFullName?: string,
    public readonly userEmail?: string,
    public readonly userAvatarUrl?: string,
  ) {}

  static fromJson(json: any): OrganizationMemberModel {
    const rawProfiles = json.profiles || json.user || {};
    return new OrganizationMemberModel(
      json.id,
      json.organization_id || json.organizationId,
      json.user_id || json.userId,
      (json.role || 'staff') as OrganizationRole,
      (json.status || 'active') as MemberStatus,
      json.created_at || json.createdAt || new Date().toISOString(),
      json.updated_at || json.updatedAt || new Date().toISOString(),
      rawProfiles.full_name || rawProfiles.fullName || json.user_full_name || json.userFullName,
      rawProfiles.email || json.user_email || json.userEmail,
      rawProfiles.avatar_url || rawProfiles.avatarUrl || json.user_avatar_url || json.userAvatarUrl,
    );
  }

  toJson(): Record<string, any> {
    return {
      id: this.id,
      organization_id: this.organizationId,
      user_id: this.userId,
      role: this.role,
      status: this.status,
      created_at: this.createdAt,
      updated_at: this.updatedAt,
      user_full_name: this.userFullName,
      user_email: this.userEmail,
      user_avatar_url: this.userAvatarUrl,
    };
  }
}
